docs/aws/audit/redshiftmonitoring/rules/redshift_cluster_in_vpc.mdx

```diff
@@ -23,6 +23,206 @@ CBP
+1. Sign in to the AWS Management Console.
+2. Navigate to the Redshift dashboard by selecting "Services" from the top menu, then selecting "Redshift" under the "Database" category.
+3. In the Redshift dashboard, select "Clusters" from the left-hand menu. This will display a list of all your Redshift clusters.
+4. For each cluster, check the "VPC ID" column. If a cluster has a VPC ID listed, it is launched within a VPC. If the "VPC ID" column is blank or not listed, the cluster is not launched within a VPC, indicating a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all the Redshift clusters available in your AWS account. You can do this by using the AWS CLI command:
+
+```bash
+aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier]' --output text
+```
+
+This command will return a list of all Redshift cluster identifiers.
+
+2. Next, for each cluster identifier returned from the previous step, you need to describe the cluster to get its details. You can do this by using the AWS CLI command:
+
+```bash
+aws redshift describe-clusters --cluster-identifier <ClusterIdentifier> --query 'Clusters[*].[VpcId]' --output text
+```
+
+Replace `<ClusterIdentifier>` with the actual cluster identifier. This command will return the VPC ID in which the Redshift cluster is launched.
+
+3. If the VPC ID returned from the previous step is `None` or `null`, it means that the Redshift cluster is not launched within a VPC. This is a misconfiguration as Redshift clusters should always be launched within a VPC for better security and isolation.
+
+4. Repeat the above steps for all Redshift clusters in your AWS account. If any cluster is not launched within a VPC, it is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) in your local environment. This will allow you to interact with AWS services including Redshift.
+
+```python
+pip install boto3
+aws configure
+```
+
+2. Import the necessary modules and create a session using your AWS credentials.
+
+```python
+import boto3
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+```
+
+3. Create a Redshift client using the session.
+
+```python
+redshift = session.client('redshift')
+```
+
+4. Use the `describe_clusters` method to retrieve information about all Redshift clusters. Then, iterate over the clusters and check the `VpcId` attribute. If a cluster does not have a `VpcId`, it means it is not launched within a VPC.
+
+```python
+clusters = redshift.describe_clusters()
+for cluster in clusters['Clusters']:
+    if 'VpcId' not in cluster:
+        print(f"Cluster {cluster['ClusterIdentifier']} is not launched within a VPC.")
+```
+
+This script will print the identifiers of all Redshift clusters that are not launched within a VPC.
+</Accordion>
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>